src/query/storages/hive/src/hive_catalog.rs

```diff
@@ -81,6 +82,42 @@ impl HiveCatalog {
         Ok(ThriftHiveMetastoreSyncClient::new(i_prot, o_prot))
     }
 
+    pub async fn get_partitions_async(
+        &self,
+        db: String,
+        table: String,
+        partition_names: Vec<String>,
+    ) -> Result<Vec<Partition>> {
+        let client = self.get_client()?;
+        tokio::task::spawn_blocking(move || {
+            Self::do_get_partitions(client, db, table, partition_names)
+        })
+        .await
+        .unwrap()
```

Comment:
```suggestion
.await?
```

Reply:
if using '?' ,   some errors:
```
the trait `From<JoinError>` is not implemented for `ErrorCode`
```

Reply:
is it nesseserry to  convert JoinError to errorCode?

Reply:
No need. 
Please keep it.